applications/StructuralMechanicsApplication/custom_strategies/custom_strategies/eigensolver_strategy.hpp

```diff
@@ -669,6 +680,34 @@ class EigensolverStrategy
                 }
         }
     }
+    ///
+     /**
+     * Computes the modal decomposition depending on the number of eigenvalues
+     * chosen and stores them in the corresponding variables. Can be activated by setting
+     * bool variable exposed to the python interface.
+     */
+    void ComputeModalDecomposition(DenseMatrixType& rEigenvectors)
+    {
+        SparseMatrixType& rMassMatrix = this->GetMassMatrix();
+        SparseMatrixType m_temp = ZeroMatrix(rEigenvectors.size1(),rEigenvectors.size2());
+        boost::numeric::ublas::axpy_prod(rEigenvectors,rMassMatrix,m_temp,true);
+        Matrix modal_mass_matrix = ZeroMatrix(m_temp.size1(),m_temp.size1());
+        boost::numeric::ublas::axpy_prod(m_temp,trans(rEigenvectors),modal_mass_matrix);
+
+        SparseMatrixType& rStiffnessMatrix = this->GetStiffnessMatrix();
+        SparseMatrixType k_temp = ZeroMatrix(rEigenvectors.size1(),rEigenvectors.size2());
+        boost::numeric::ublas::axpy_prod(rEigenvectors,rStiffnessMatrix,k_temp,true);
+        Matrix modal_stiffness_matrix = ZeroMatrix(k_temp.size1(),k_temp.size1());
+        boost::numeric::ublas::axpy_prod(k_temp,trans(rEigenvectors),modal_stiffness_matrix);
```

Comment:
Once we move to AMatrix this might not be working for some time, just saying since I am not sure if this is already available there